src/sentry/api/event_search.py

```diff
@@ -531,6 +576,27 @@ def convert_term(term):
     return [operator, [left, right]]
 
 
+def convert_result_filter_to_snuba_query(result_filter):
+    name = result_filter.key.name
+    value = result_filter.value.value
+
+    value = (
+        int(to_timestamp(value)) * 1000
+        if isinstance(value, datetime) and name != "timestamp"
+        else value
+    )
+
+    if result_filter.operator in ("=", "!=") and result_filter.value.value == "":
+        return [["isNull", [name]], result_filter.operator, 1]
+
+    _, agg_additions = resolve_field(name)
+    if len(agg_additions) > 0:
+        name = agg_additions[0][-1]
+
+    condition = [name, result_filter.operator, value]
```

Comment:
This looks like it would enable people to do having conditions on fields they didn't select. That will enable them to add implicit aggregations to their query and create un-expected results. For example if the selected columns are `transaction, project, timestamp` and the user adds a condition of `p95:>1000` we have to aggregate the results to satisfy the `p95` but that the results don't have any indication that they've been aggregated.

I would only allow conditions on aggregates that are in the selected fields to keep our lives simpler.

Reply:
We already do this with missing group by fields don't we? I thought we were already implicitly modifying people's queries.

Reply:
I ended up adding this check, just because we needed to check for the unsupported aliases right now too.